index.js

```diff
@@ -58,10 +58,21 @@ class TemplateCompiler extends Filter {
   }
 
   processString(string, relativePath) {
-    return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
-      contents: string,
-      moduleName: relativePath
-    }) + ';';
+    try {
+      return 'export default ' + utils.template(this.options.templateCompiler, stripBom(string), {
+        contents: string,
+        moduleName: relativePath
+      }) + ';';
+    } catch(e) {
+      // Create a new error, augment it with location and type information
+      // and re-throw.
+      let error = new Error(e);
```

Comment:
I don't think `broccoli-persistent-filter` wraps in `try-catch` but maybe I'm wrong? [looking here](https://github.com/stefanpenner/broccoli-persistent-filter/blob/master/index.js#L353-L357).

as far as why not attaching properties to existing error -- my thinking was that we're going to use one `BuildError` class to communicate exceptions and would ideally want to just `throw new BuildError()` w/o add-on developers knowing too much about which properties need to be attached. Having said that, in this particular case, we don't necessarily need to create a new `Error` object and can re-throw after attaching info we need.